worldedit-core/src/main/java/com/fastasyncworldedit/core/util/MathMan.java

```diff
@@ -152,52 +152,68 @@ public static long pairInt(int a, int b) {
     }
 
     public static long tripleWorldCoord(int x, int y, int z) {
-        return y + (((long) x & 0x3FFFFFF) << 8) + (((long) z & 0x3FFFFFF) << 34);
+        return ((y + 256) & 0xffff) + (((long) x & 0xffffff) << 16) + (((long) z & 0xffffff) << 40);
     }
 
     public static long untripleWorldCoordX(long triple) {
-        return (((triple >> 8) & 0x3FFFFFF) << 38) >> 38;
+        return (((triple >> 16) & 0xffffff) << 38) >> 38;
     }
 
     public static long untripleWorldCoordY(long triple) {
-        return triple & 0xFF;
+        return (triple & 0xffff) - 256;
     }
 
     public static long untripleWorldCoordZ(long triple) {
-        return (((triple >> 34) & 0x3FFFFFF) << 38) >> 38;
+        return (((triple >> 40) & 0xffffff) << 38) >> 38;
     }
 
-    public static short tripleBlockCoord(int x, int y, int z) {
-        return (short) ((x & 15) << 12 | (z & 15) << 8 | y);
+    public static int tripleBlockCoord(int x, int y, int z) {
+        // account for the fact y can be negative now. Assume it won't be less than -256
+        y += 256;
+        return ((x & 15) << 16 | (z & 15) << 12 | y);
     }
 
     public static char tripleBlockCoordChar(int x, int y, int z) {
-        return (char) ((x & 15) << 12 | (z & 15) << 8 | y);
+        return (char) ((x & 15) << 16 | (z & 15) << 12 | y);
     }
 
     public static int untripleBlockCoordX(int triple) {
-        return (triple >> 12) & 0xF;
+        return (triple >> 16) & 0xF;
     }
 
     public static int untripleBlockCoordY(int triple) {
-        return (triple & 0xFF);
+        return (triple & 0x1ff) - 256;
     }
 
     public static int untripleBlockCoordZ(int triple) {
-        return (triple >> 8) & 0xF;
+        return (triple >> 12) & 0xF;
     }
 
+    /**
+     * Obtain an integer representation of 3 ints, x, y and z. y is represented by the right-most 9 bits and can
+     * be within the range -256 to 255. x and z are represented by 11 bits each and can be within the range
+     * -1024 to 1024
```

Comment:
-1024 to 1023, and we can use 10 bits for y (see my comment below, I assumed 10 bits there)